tests/test_utils/test_chroma_store.py

```diff
@@ -0,0 +1,161 @@
+"""Test suite for testing chroma_store utility."""
+import tempfile
+
+import chromadb
+import pytest
+from chromadb.api import API
+from chromadb.api.types import Documents, EmbeddingFunction, Embeddings
+from chromadb.config import Settings
+from utils.chroma_store import ChromaStore
+
+
+@pytest.fixture
+def local_persist_api() -> API:
+    """Fixture to create a local chroma server for testing.
+
+    Returns:
+        API: Local chroma server
+    """
+    return chromadb.Client(
+        Settings(
+            chroma_api_impl="local",
+            chroma_db_impl="duckdb+parquet",
+            persist_directory=tempfile.gettempdir() + "/test_server",
+        )
+    )
+
+
+class FakeEmbeddingFunction(EmbeddingFunction):
+    """Fake embedding class for testing."""
+
+    def __call__(self, texts: Documents) -> Embeddings:
+        """Calls the fake embedding function.
+
+        Args:
+            texts (Documents): Documents to embed
+
+        Returns:
+            Embeddings: Return fixed embedding
+        """
+        return [[float(1.0)] * 9 + [float(i)] for i in range(len(texts))]
+
+
+@pytest.mark.parametrize(
+    "collection_name, is_valid",
+    [
+        ("areallylongnameforcollectionname" * 3, False),
+        ("a..b", False),
+        ("Test_b", False),
+        ("ba_tesT", False),
+        ("192.124.0.1", False),
+        ("AbbA", False),
+        ("a_valid-name", True),
+    ],
+)
+def test_validate_collection_name(collection_name: str, is_valid: bool):
+    """Test if collection names are valid.
+
+    Args:
+        collection_name (str): Name of collection
+        is_valid (bool): Whether the input collection name is valid
+    """
+    store = ChromaStore()
+    store._client = local_persist_api
+    validation = store.validate_collection_name(collection_name)
+    assert validation.is_valid == is_valid
+
+
+def test_get_or_create_collection(local_persist_api: API):
+    """Test creating multiple collections in chromadb using `_get_or_create_collection` function.
+
+    Args:
+        local_persist_api (API): Local chroma server for testing
+    """
+    store = ChromaStore()
+    store._client = local_persist_api
+
+    # Create collection "test"
+    store._get_or_create_collection("test", FakeEmbeddingFunction())
+    assert store._collection.name == "test"
+    assert "test" in store.list_collection_names()
+
+    # Create collection "test1"
+    store._get_or_create_collection("test1", FakeEmbeddingFunction())
+    assert store._collection.name == "test1"
+    assert "test1" in store.list_collection_names()
```

Comment:
Nit: 

I would do:

```python
assert {'test', 'test1'} == set(store.list_collection_names())
```

As you also want to make sure that the previously created collection is still there.